### ISSUE TITLE
Add relative and absolute options to all MIDI controls

### ISSUE THREAD

**Issue by latenitefilms** [fcpxhacks/fcpxhacks#1673]
‎Al Piazza [writes](https://www.facebook.com/groups/commandpost/permalink/2222654891323543/):

> Sorry chris one other thing . When mapping a slider (volume controller) on a midi device I noticed that only a single value is applied.
> Example: when I use “set”and move the slider only a single value is set in the value field. So if a wanted to use it as a volume control it wouldn’t work. Need to have a sort of increase and decrease type of value. 
> This would be really useful on things like increasing and decreasing clip height as you would be able to use a midi controllers volume slider to change clip height by just sliding it up and down. The slider does have a set limit of 0 to 127. I understand that it is hard to know what the midi volume slider (or any controller) is set to at any giving time but other software use a thing they call “pickup” 
> So if the volume controller is at 50 and the value to change is at 70 and you move the volume control it does nothing to the value until it hits 70 and then it “picks up” and starts controlling. Again my explanation may be hard to understand.

**Comment by MrTechGadget** [comment]
Any chance this got implemented and just not closed out here? I am relatively new to Command Post and can't figure out how to get a slider mapped to volume. I'm used to this working in OBS and would love to have this work in many other apps, like FCP, Logic, and just the overall System Volume in all apps.

**Comment by latenitefilms** [comment]
It really depends on your MIDI device. A lot of MIDI devices can send different messages if you turn a knob left vs right, etc. so which case all our existing actions work. We've added SOME specific MIDI actions that take into account relative/absolute commands - but let me know if there's anything specific missing.

**Comment by MrTechGadget** [comment]
Yeah, my faders send MIDI control commands with a value of 0-127. I am only seeing an event fire at that specific value.
Nothing I can find that will map a minimum and maximum control (like volume) over the range of values.
